app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/relationships/RelationshipPresenter.kt

```diff
@@ -51,47 +57,69 @@ class RelationshipPresenter internal constructor(
             .withTrackedEntityAttributeValues()
             .uid(teiUid)
             .blockingGet()?.trackedEntityType()
-    var updateRelationships: FlowableProcessor<Boolean> = PublishProcessor.create()
+    private var updateRelationships: FlowableProcessor<Boolean> = PublishProcessor.create()
 
-    private val _relationshipsModels = MutableLiveData<List<RelationshipViewModel>>()
-    val relationshipModels: LiveData<List<RelationshipViewModel>> = _relationshipsModels
+    private val _relationshipsModels = MutableLiveData<List<RelationshipModel>>()
     private val _relationshipMapData: MutableLiveData<RelationshipMapData> = MutableLiveData()
     val relationshipMapData: LiveData<RelationshipMapData> = _relationshipMapData
     private val _mapItemClicked = MutableLiveData<String>()
     val mapItemClicked: LiveData<String> = _mapItemClicked
+    private val job = Job()
+    private val scope = CoroutineScope(dispatcherProvider.ui() + job)
 
     fun init() {
-        compositeDisposable.add(
-            updateRelationships.startWith(true)
-                .flatMapSingle { relationshipRepository.relationships() }
-                .subscribeOn(schedulerProvider.io())
-                .observeOn(schedulerProvider.ui())
-                .subscribe(
-                    {
-                        _relationshipsModels.postValue(it)
-                    },
-                    { Timber.d(it) },
-                ),
-        )
+        scope.launch {
+            relationshipsRepository.getRelationships().collect {
```

Comment:
This could be a potential leak source. It would be better to something like:

`relationshipsRepository.getRelationships.onEach(::doWhatever).launchIn(scope)`

It would be better if the presenter was a viewmodel, as it will tie the collection with the lifecycle and dispose it automatically if needed. I think right now the job has to be manually cancelled to avoid any problem.